operators/multiclusterobservability/controllers/placementrule/placementrule_controller.go

```diff
@@ -514,44 +516,74 @@ func updateManagedClusterList(obj client.Object) {
 	}
 }
 
+// Do not reconcile objects if this instance of mch has the
+// `disableAddonAutomaticInstallationAnnotationKey` annotation
+func isAutomaticAddonInstallationDisabled(obj client.Object) bool {
+	annotations := obj.GetAnnotations()
+	if val, ok := annotations[disableAddonAutomaticInstallationAnnotationKey]; ok && strings.EqualFold(val, "true") {
+		log.Info("Cluster has disable addon automatic installation annotation. Skip addon deploy")
+		return true
+	}
+	return false
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *PlacementRuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	c := mgr.GetClient()
 	ingressCtlCrdExists := r.CRDMap[config.IngressControllerCRD]
 	clusterPred := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
 			log.Info("CreateFunc", "managedCluster", e.Object.GetName())
+
 			updateManagedClusterList(e.Object)
 			updateManagedClusterImageRegistry(e.Object)
+
+			if isAutomaticAddonInstallationDisabled(e.Object) {
+				return false
+			}
+
```

Comment:
Could this be moved to the top of Create() as an early return?

Reply:
yes, you need to do that to avoid updateManagedClusterList() method invoked when disable annotations is added